internal/signalfx-agent/pkg/utils/filter/filter.go

```diff
@@ -21,6 +21,7 @@ type StringFilter interface {
 // StringMapFilter matches against the values of a map[string]string.
 type StringMapFilter interface {
 	Matches(map[string]string) bool
+	MatchesAny(map[string]any) bool
```

Comment:
Rather than using raw maps, what about accepting `pcommon.Map` so we can the `Range` and value `AsString()` helpers?